### ISSUE TITLE
Image Block is missing 'srcset' attribute in the editor

### ISSUE THREAD

**Issue by danielbachhuber** [WordPress/gutenberg#46953]
When I insert an image into the editor and 'large' is set as my default size (https://github.com/WordPress/gutenberg/pull/29966), the image appears fuzzy:

<img width="1436" alt="image" src="https://user-images.githubusercontent.com/36432/211086758-2796cf5f-9e5b-4200-83a2-f3ce357a3643.png">

It seems to be missing its `sizes` and `srcset` attributes. If we added them, it would appear much sharper.

Also, it's interesting that we let the user choose and image size _at all_ in this scenario. Given WordPress supports responsive images and width/height controls, image sizes are totally irrelevant.

**Done is:**
* The `sizes` and `srcset` attributes are present for images in the Image Block.

**Comment by skorasaurus** [comment]
I never noticed that the srcset and sizes attributes for images were omitted on the editor side; good observation; confirmed.

**Comment by danielbachhuber** [comment]
> I never noticed that the srcset and sizes attributes for images were omitted on the editor side.

I imagine there's some technical complexity to calculate them. I haven't evaluated at depth, however.

**Comment by mtias** [comment]
Yeah, some of this is a bit legacy behavior that we should revisit. Originally it wasn't a top priority to use srcset in the editor context compared to the front-end while ensuring all manipulations happened without UX hiccups (cropping and so on). Would be good to revise where things stand now.

**Comment by Mamaduka** [comment]
It would be nice to revisit this. @Vedant-Gandhi started working on a PR.

 @felixarntz, any suggestions based on your experience from working on similar resolutions in the PHP part?
